torch/csrc/autograd/engine.cpp

```diff
@@ -371,6 +400,7 @@ void Engine::reentrant_thread_init() {
       continue;
     }
     set_device(graph_task->owner_);
+    local_ready_queue = parent_ready_queue;
```

Comment:
When the reentrant thread finishes executing, we should unset `local_ready_queue`, right? (And then you can assert, at entry, that `TORCH_INTERNAL_ASSERT(!local_ready_queu)`